WooCommerce/Classes/ViewRelated/Orders/Order Creation/ProductsSection/ProductRowViewModel.swift

```diff
@@ -49,14 +49,19 @@ final class ProductRowViewModel: ObservableObject, Identifiable {
     ///
     private let manageStock: Bool
 
-    /// Label showing product details: stock status, price, and variations (if any).
+    /// Product variation attributes
+    ///
+    private let variationAttributes: [VariationAttributeViewModel]?
+
+    /// Label showing product details. Can include stock status or attributes, price, and variations (if any).
     ///
     var productDetailsLabel: String {
-        let stockLabel = createStockText()
+        // When provided, the variation attributes should replace the stock status
```

Comment:
Rather than have this implicit logic, wouldn't it be better if we provided a proper type on the `init`? Something along the lines of 
```swift
enum VariationDisplayMode {
    case stock
    case attributes(Attributes)
}
```

Reply:
That's a great idea; it feels much better to set that explicitly. I added that in [5c2c923](https://github.com/woocommerce/woocommerce-ios/pull/6131/commits/5c2c9239d2a0465b1f6682503b558d6fb2395028), so that initializing with a product variation sets a display mode that determines which text to display, and otherwise (for non-variations) it defaults to the stock status text.